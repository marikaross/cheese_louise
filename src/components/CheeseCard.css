.CheeseCard {
  width: 430px;
  height: 200px;
  background-image: url('../images/low-contrast-linen.png');
  color: #fefaea;
  margin: 10px;
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.6);
}

.overlay {
  background-color: rgba(54, 74, 117, .95);
  width: 430px;
  height: 200px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  justify-content: space-around;
}
.cheese-image {
  width: 100%;
  height: 180px;
  -webkit-filter: grayscale(.4);
  filter: grayscale(.4);
  margin: 10px;
  box-shadow: 0px 2px 10px 0px rgba(255, 254, 254, 0.6);
}

.cheese-info {
  flex-direction: column;
  padding-left: 20px;
  padding-right: 20px;
}

h2 {
  font-size: 1.4em;
}

h3 {
  font-size: 1em;
}

.CheeseCard:hover {
  transform: scale(1.01);
}

.details-link {
  text-decoration: none;
  color: #e9eae4;
  display: block;
}

button {
  background-color: #fefaea;
  border: none;
  color: #ba3e30;
  font-size: 1.1em;
  margin: 5px;
  border-radius: 7%;
  box-shadow: 0px 2px 5px 0px rgba(255, 254, 254, 0.8);

}

button:hover {
  cursor: pointer;
  /*color*/
}

.cheese-icon {
  width: 100%;
  height: 235px;
}

@media screen and (max-width: 920px) {
  .CheeseCard,
  .overlay {
    width: 380px;
  }

}

@media screen and (max-width: 800px) {
  .CheeseCard,
  .overlay {
    width: 300px;
  }

  .cheese-icon{
    margin-left: 5px
  }
}

@media screen and (max-width: 650px) {
  .Cheesecard,
  .overlay {
    width: 400px;
  }
}

@media screen and (max-width: 420px) {
  .Cheesecard,
  .overlay {
    width: 380px;
  }
}

@media screen and (max-width: 405px) {
  .Cheesecard,
  .overlay {
    width: 300px;
  }
}

