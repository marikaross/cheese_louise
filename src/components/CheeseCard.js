import React from 'react';
import './CheeseCard.css';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { fetchSummary } from '../thunks/fetchSummary';
import PropTypes from 'prop-types';


export const CheeseCard = (props) => {

  const makeUrl = () => {
    const id = props.id
    const urlSnippet = props.frenchWiki
    props.handleFetch(id, urlSnippet)
  }

  return (
    <article className='CheeseCard' key={props.id}>
      <img className='cheese-image' src={`https://public.opendatasoft.com/explore/dataset/frenchcheese/files/${props.picture}/300/`} />
      <section className='cheese-info'>
        <h2>{props.name}</h2>
        <button onClick={() => props.toggleFave(props.id)}>favorite</button>
        <Link className='details-link'to={`/${props.id}`} onClick={() => makeUrl()}>See more info</Link>
      </section>
    </article>
    )
}

CheeseCard.propTypes = {
  id: PropTypes.string,
  picture: PropTypes.string,
  name: PropTypes.string,
  milk: PropTypes.string,
  region: PropTypes.string
}
