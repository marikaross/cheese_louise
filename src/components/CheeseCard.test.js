import { CheeseCard } from './CheeseCard';
import React from 'react';
import { shallow, mount } from 'enzyme';

describe('CheeseCard', () => {

  it('should match the snapShot', () => {
    let wrapper = shallow(<CheeseCard favorites={[{}, {}]} picture={'photo of cheese'}/>)
    expect(wrapper).toMatchSnapshot()
  })


  it('should call makeUrl when the link is clicked', () => {
    let mockMakeUrl = jest.fn()
    let mockHandleFetch = jest.fn()
    let mockCheeses = [{name: 'nacho', id: 6}, {name: 'swiss', id: 5}]
    let mockFavorites = [{name: 'nacho', id: 6}]
    let mockAddFaveCheese = jest.fn()
    let mockDeleteCheese = jest.fn()
    let mockFetchSummary = jest.fn()
    let mockToggleFave = jest.fn()
    let mockHistory = []
    let mockIsDuplicate = jest.fn()

    let wrapper = shallow( <CheeseCard 
        cheeses={mockCheeses}
        favorites={mockFavorites}
        addFaveCheese={mockAddFaveCheese}
        deleteCheese={mockDeleteCheese}
        toggleFave={mockToggleFave}
        history={[]}
        makeUrl={mockMakeUrl}
        handleFetch={mockHandleFetch}
        picture={'picture'}
      />)
    wrapper.find('.details-link').simulate('click')
    expect(makeUrl).toBeCalled()
  })
})