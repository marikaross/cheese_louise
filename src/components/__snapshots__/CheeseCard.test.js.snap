// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheeseCard should match the snapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="cheese-image"
          src="https://public.opendatasoft.com/explore/dataset/frenchcheese/files/undefined/300/"
/>,
        <section
          className="cheese-info"
>
          <h2 />
          <button
                    className="fave-button"
                    onClick={[Function]}
          >
                    favorite
          </button>
          <Link
                    className="details-link"
                    onClick={[Function]}
                    replace={false}
                    to="/undefined"
          >
                    See more info
          </Link>
</section>,
      ],
      "className": "CheeseCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "cheese-image",
          "src": "https://public.opendatasoft.com/explore/dataset/frenchcheese/files/undefined/300/",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <button
              className="fave-button"
              onClick={[Function]}
>
              favorite
</button>,
            <Link
              className="details-link"
              onClick={[Function]}
              replace={false}
              to="/undefined"
>
              See more info
</Link>,
          ],
          "className": "cheese-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "favorite",
              "className": "fave-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "favorite",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "See more info",
              "className": "details-link",
              "onClick": [Function],
              "replace": false,
              "to": "/undefined",
            },
            "ref": null,
            "rendered": "See more info",
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="cheese-image"
            src="https://public.opendatasoft.com/explore/dataset/frenchcheese/files/undefined/300/"
/>,
          <section
            className="cheese-info"
>
            <h2 />
            <button
                        className="fave-button"
                        onClick={[Function]}
            >
                        favorite
            </button>
            <Link
                        className="details-link"
                        onClick={[Function]}
                        replace={false}
                        to="/undefined"
            >
                        See more info
            </Link>
</section>,
        ],
        "className": "CheeseCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "cheese-image",
            "src": "https://public.opendatasoft.com/explore/dataset/frenchcheese/files/undefined/300/",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <button
                className="fave-button"
                onClick={[Function]}
>
                favorite
</button>,
              <Link
                className="details-link"
                onClick={[Function]}
                replace={false}
                to="/undefined"
>
                See more info
</Link>,
            ],
            "className": "cheese-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "favorite",
                "className": "fave-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "favorite",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "See more info",
                "className": "details-link",
                "onClick": [Function],
                "replace": false,
                "to": "/undefined",
              },
              "ref": null,
              "rendered": "See more info",
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
